recipes/itk/all/conanfile.py

```diff
@@ -26,269 +31,236 @@ class ITKConan(ConanFile):
+        tc.variables["Module_ITKIntegratedTest"] = True
+        tc.variables["Module_ITKKLMRegionGrowing"] = True
+        tc.variables["Module_ITKLabelMap"] = True
+        tc.variables["Module_ITKLabelVoting"] = True
+        tc.variables["Module_ITKLevelSets"] = True
+        tc.variables["Module_ITKLevelSetsv4"] = True
+        tc.variables["Module_ITKMarkovRandomFieldsClassifiers"] = True
+        tc.variables["Module_ITKMathematicalMorphology"] = True
+        tc.variables["Module_ITKMetricsv4"] = True
+        tc.variables["Module_ITKNarrowBand"] = True
+        tc.variables["Module_ITKNeuralNetworks"] = True
+        tc.variables["Module_ITKOptimizers"] = True
+        tc.variables["Module_ITKOptimizersv4"] = True
+        tc.variables["Module_ITKPDEDeformableRegistration"] = True
+        tc.variables["Module_ITKPath"] = True
+        tc.variables["Module_ITKPolynomials"] = True
+        tc.variables["Module_ITKQuadEdgeMeshFiltering"] = True
+        tc.variables["Module_ITKRegionGrowing"] = True
+        tc.variables["Module_ITKRegistrationCommon"] = True
+        tc.variables["Module_ITKRegistrationMethodsv4"] = True
+        tc.variables["Module_ITKReview"] = True
+        tc.variables["Module_ITKSignedDistanceFunction"] = True
+        tc.variables["Module_ITKSmoothing"] = True
+        tc.variables["Module_ITKSpatialFunction"] = True
+        tc.variables["Module_ITKTBB"] = True
+        tc.variables["Module_ITKThresholding"] = True
+        tc.variables["Module_ITKVideoCore"] = True
+        tc.variables["Module_ITKVideoFiltering"] = True
+        tc.variables["Module_ITKVideoIO"] = False
+        tc.variables["Module_ITKVoronoi"] = True
+        tc.variables["Module_ITKWatersheds"] = True
+        tc.variables["Module_ITKDICOMParser"] = True
+
+        tc.variables["Module_ITKVTK"] = False
+        tc.variables["Module_ITKVtkGlue"] = False
 
         # Disabled on Linux (link errors)
-        self._cmake.definitions["Module_ITKLevelSetsv4Visualization"] = False
+        tc.variables["Module_ITKLevelSetsv4Visualization"] = False
 
         # Disabled because Vxl vidl is not built anymore
-        self._cmake.definitions["Module_ITKVideoBridgeVXL"] = False
+        tc.variables["Module_ITKVideoBridgeVXL"] = False
 
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+        tc.generate()
+
+        tc = CMakeDeps(self)
+        tc.generate()
+
+    def _patch_sources(self):
+        apply_conandata_patches(self)
+        #The CMake policy CMP0091 must be NEW, but is ''
+        replace_in_file(self,
+                        os.path.join(self.source_folder, "Modules", "ThirdParty", "VNL", "src", "vxl", "config", "cmake", "config", "VXLIntrospectionConfig.cmake"),
+                        "-DCMAKE_CXX_FLAGS:STRING=${CMAKE_CXX_FLAGS}",
+                        "-DCMAKE_POLICY_DEFAULT_CMP0091=NEW -DCMAKE_CXX_FLAGS:STRING=${CMAKE_CXX_FLAGS}")
```

Comment:
Would setting this from generate() not work? Or would it be too broadly applied, perhaps?

```python
tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0091"] = "NEW"
```

Reply:
I've tried using cache_variables before adding this replacement. It doesn't work (CMake 3.25.2).
![image](https://github.com/conan-io/conan-center-index/assets/4276548/f957d30d-446c-4860-9a9e-4cc60d2ffedb)

Reply:
@Nekto89 Have you tried with `variables["CMAKE_POLICY_DEFAULT_CMP0091"]` I had similar problem days ago, but it started to work magically, bad cache maybe. Plus, I didn't find an explicit `CMP0091` in ITK project, which means, passing by CMakeToolchaing should work

Reply:
@uilianries  I think that try_compile doesn't propagate policies but still reuses the same cmake toolchain file
https://github.com/InsightSoftwareConsortium/ITK/blob/a0125aa894b2cf7aaa0ba42a18d80c485a8fffa5/Modules/ThirdParty/VNL/src/vxl/config/cmake/config/VXLIntrospectionConfig.cmake#L114C5-L114C16

I have this error even locally. cmake policy is already automatically used by conan but it's not enough. This is the log with explicit tc.variables["CMAKE_POLICY_DEFAULT_CMP0091"] = "NEW" instead of replace_in_file.

```
-- Performing Test VXL_PIC_COMPATIBLE
CMake Error at C:/.conan/ae193e/1/build/generators/conan_toolchain.cmake:31 (message):
  The CMake policy CMP0091 must be NEW, but is ''
Call Stack (most recent call first):
  C:/.conan/ae193e/1/build/CMakeFiles/3.25.2/CMakeSystem.cmake:6 (include)
  C:/.conan/ae193e/1/src/Modules/ThirdParty/VNL/src/vxl/config/cmake/config/vxl_shared_link_test/CMakeLists.txt:14 (project)


CMake Error at Modules/ThirdParty/VNL/src/vxl/config/cmake/config/VXLIntrospectionConfig.cmake:114 (try_compile):
  Failed to configure test project build system.
Call Stack (most recent call first):
  Modules/ThirdParty/VNL/src/vxl/config/cmake/config/VXLIntrospectionConfig.cmake:402 (PERFORM_CMAKE_TEST_CUSTOM)
  Modules/ThirdParty/VNL/src/vxl/CMakeLists.txt:252 (include)


-- Configuring incomplete, errors occurred!
See also "C:/.conan/ae193e/1/build/CMakeFiles/CMakeOutput.log".
See also "C:/.conan/ae193e/1/build/CMakeFiles/CMakeError.log".
itk/5.1.2:
itk/5.1.2: ERROR: Package '9fda978a8515339375bf630436153616b0e1b073' build failed
itk/5.1.2: WARN: Build folder C:\.conan\ae193e\1\build
ERROR: itk/5.1.2: Error in build() method, line 250
        cmake.configure()
        ConanException: Error 1 while executing cmake -G "Visual Studio 17 2022" -DCMAKE_TOOLCHAIN_FILE="C:/.conan/ae193e/1/build/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="C:/.conan/68883d/1" -DCMAKE_PROJECT_ITK_INCLUDE="C:\.conan\ae193e\1\src\conan_cmake_project_include.cmake" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" "C:\.conan\ae193e\1\src"
```

Reply:
@Nekto89 thank you for checking it!